woocommerce/woocommerce-admin
# Reduce the specificity and complexity of the `ReportError` component

This is a baby step towards https://github.com/woocommerce/woocommerce-admin/issues/6243
it reduces the specificity and complexity of the component, so it would be easier to share it.

(TBH, I don't see a huge value in publishing this exact component. But this is a dependency of other `Report*` components, so would be eventually needed, and hopefully would help me pave my path to contribute to this repo ;) )

- Remove unused `isEmpty` prop from ReportError. (4ef2c61)
- Remove `isError` prop from ReportError, as it is always set. Also, to reduce confusion that: Could "report error" be not an error? (82fb25d)
- Remove default value of `className` prop from `ReportError`, it is unnecesary, as `EmptyContent` takes falsy values as well as `''`. (02d74ff)
- Change `ReportError` to a function component. (ead3313)

(I believe cleaning up / limiting the API will be beneficial, especially before publishing the package. So we would not get stuck supporting more variants than actually needed, just for the sake of backward compatibility / a chance of somebody using it somewhere)

### Accessibility




:point_up: The component behaves as it was in respect of A11y.

### Screenshots

### Detailed test instructions:

- Open a report page that shows an error, for example provide some invalid URL param like: [`/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fproducts&chart=items_sold&orderby=items_sold_____BOGUS____&order=desc`](https://woo.test/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fproducts&chart=items_sold&orderby=items_sold_____BOGUS____&order=desc)
- Check that the error behaves as before.
